input/file/inputfile.go

```diff
@@ -69,6 +69,11 @@ func InitHandler(ctx context.Context, raw *config.ConfigRaw) (config.TypeInputCo
 		return nil, err
 	}
 
+	conf.Codec, err = config.GetCodec(ctx, *raw)
```

Comment:
forget to handle error?
```
if err != nil {
    return nil, err
}
```

Reply:
Here we don't actually care about the error, because if no codec is configured, the internal mapping will cause an error, but it is not relevant (IMHO).  If there is no Codec, for one reason, or another, we just use the default, no?